src/v2/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -0,0 +1,168 @@
+import { createFragmentContainer, graphql } from "react-relay"
+import { FC, useState } from "react"
+import { BankDebitProvider } from "v2/Components/BankDebitForm/BankDebitProvider"
+import { BankAccountPicker_me } from "v2/__generated__/BankAccountPicker_me.graphql"
+import {
+  BorderedRadio,
+  RadioGroup,
+  Spacer,
+  Collapse,
+  Button,
+} from "@artsy/palette"
+import { BankDebitDetails } from "./BankDebitDetails"
+import { BankAccountPicker_order } from "v2/__generated__/BankAccountPicker_order.graphql"
+import { useSetPayment } from "v2/Apps/Order/Components/Mutations/useSetPayment"
+
+interface Props {
+  order: BankAccountPicker_order
+  me: BankAccountPicker_me
+  onSetPaymentSuccess: () => void
+  onSetPaymentError: (error: Error) => void
+}
+
+export const BankAccountPicker: FC<Props> = props => {
+  const getInitialBankAccountSelection = () => {
+    if (props.order.bankAccountId) {
+      return {
+        type: "existing",
+        id: props.order.bankAccountId,
+      }
+    } else {
+      return props.me.bankAccounts?.edges?.length
+        ? {
+            type: "existing",
+            id: props.me.bankAccounts.edges[0]?.node?.internalID!,
+          }
+        : { type: "new" }
+    }
+  }
+
+  const [bankAccountSelection, setBankAccountSelection] = useState(
+    getInitialBankAccountSelection()
+  )
+
+  const {
+    me: { bankAccounts },
+    order,
+    onSetPaymentSuccess,
+    onSetPaymentError,
+  } = props
+
+  const bankAccountsArray = bankAccounts?.edges?.map(e => e?.node)!
+
+  const userHasExistingBankAccounts = bankAccountsArray.length > 0
+
+  const { submitMutation: setPaymentMutation } = useSetPayment()
+
+  const handleContinue = async () => {
+    try {
+      const orderOrError = (
+        await setPaymentMutation({
+          variables: {
+            input: {
+              id: order.internalID,
+              paymentMethod: "US_BANK_ACCOUNT",
+              paymentMethodId: bankAccountSelection.id,
+            },
+          },
+        })
+      ).commerceSetPayment?.orderOrError
+
+      if (orderOrError?.error) {
+        throw orderOrError.error
+      }
+
+      onSetPaymentSuccess()
+    } catch (error) {
+      onSetPaymentError(error)
+    }
+  }
+
+  return (
+    <>
+      {userHasExistingBankAccounts && (
+        <RadioGroup
+          onSelect={val => {
```

Comment:
#non-blocking. I feel the rendering here is a bit hard to follow because of all the inline logic. Perhaps it reads better if we move the logic to some functions and the rendering would look more concise.